sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkDoubleUpDownCounter.java

```diff
@@ -82,25 +91,30 @@ public DoubleUpDownCounterBuilder setAdvice(
 
     @Override
     public DoubleUpDownCounter build() {
-      return buildSynchronousInstrument(SdkDoubleUpDownCounter::new);
+      return builder.buildSynchronousInstrument(SdkDoubleUpDownCounter::new);
     }
 
     @Override
     public ObservableDoubleUpDownCounter buildWithCallback(
         Consumer<ObservableDoubleMeasurement> callback) {
-      return registerDoubleAsynchronousInstrument(
+      return builder.buildDoubleAsynchronousInstrument(
           InstrumentType.OBSERVABLE_UP_DOWN_COUNTER, callback);
     }
 
     @Override
     public ObservableDoubleMeasurement buildObserver() {
-      return buildObservableMeasurement(InstrumentType.OBSERVABLE_UP_DOWN_COUNTER);
+      return builder.buildObservableMeasurement(InstrumentType.OBSERVABLE_UP_DOWN_COUNTER);
     }
 
     @Override
     public DoubleUpDownCounterAdviceConfigurer setAttributes(List<AttributeKey<?>> attributes) {
-      adviceBuilder.setAttributes(attributes);
+      builder.setAdviceAttributes(attributes);
       return this;
     }
+
+    @Override
+    public String toString() {
```

Comment:
Missing tests for all of these?

Reply:
Are you suggesting that we should have tests for `toString()` methods?

Reply:
We do for some (most?) of the `toString()` methods; perhaps we should add them for the instrument builders as well.

Reply:
Hmmm, I've traditionally been hesitant to test `toString()` because it's supposed to be purely informative and relying on `toString()` for real programmatic concerns is probably a bad, fragile practice. 

In this case, I might also shy away from testing it because the implementation relies on the `toString()` of several other classes, including autovalue generated `toString()` from the `InstrumentDescriptor`. Is that really that helpful?